cvat/apps/consensus/merge_consensus_jobs.py

```diff
@@ -0,0 +1,181 @@
+    define_dependent_job,
+    get_rq_job_meta,
+    get_rq_lock_by_user,
+    process_failed_job,
+)
+from cvat.apps.quality_control.quality_reports import JobDataProvider
+
+
+def get_consensus_jobs(task_id: int) -> Tuple[Dict[int, List[Tuple[int, User]]], List[Job]]:
+    jobs = {}  # parent_job_id -> [(consensus_job_id, assignee)]
+
+    for job in Job.objects.select_related("segment").filter(
+        segment__task_id=task_id, type=JobType.CONSENSUS.value
+    ):
+        assert job.parent_job_id
+
+        # if the job is in NEW state, it means that the job isn't annotated
+        if job.state == StateChoice.NEW.value:
+            continue
+        jobs.setdefault(job.parent_job_id, []).append((job.id, job.assignee))
+
+    parent_job_ids = list(jobs.keys())
+
+    parent_jobs: List[Job] = []
+    # remove parent jobs that are not in annotation stage
+    for parent_job_id in parent_job_ids:
+        parent_job = Job.objects.filter(id=parent_job_id, type=JobType.ANNOTATION.value).first()
+
+        if parent_job.stage == StageChoice.ANNOTATION.value:
+            parent_jobs.append(parent_job)
+        else:
+            jobs.pop(parent_job_id)
+
+    return jobs, parent_jobs
+
+
+def get_annotations(job_id: int) -> dm.Dataset:
+    return JobDataProvider(job_id).dm_dataset
+
+
+@transaction.atomic
+def _merge_consensus_jobs(task_id: int) -> None:
+    jobs, parent_jobs = get_consensus_jobs(task_id)
+    if not jobs:
+        raise ValidationError(
+            "No annotated consensus jobs found or no regular jobs in annotation stage"
+        )
+
+    consensus_settings = ConsensusSettings.objects.filter(task=task_id).first()
+    merger = IntersectMerge(
+        conf=IntersectMerge.Conf(
+            pairwise_dist=consensus_settings.iou_threshold,
+            output_conf_thresh=consensus_settings.agreement_score_threshold,
+            quorum=consensus_settings.quorum,
+            sigma=consensus_settings.sigma,
+        )
+    )
+
+    job_comparison_reports: Dict[int, ComparisonReport] = {}
+
+    for parent_job_id, consensus_job_info in jobs.items():
+        consensus_job_ids = [consensus_job_id for consensus_job_id, _ in consensus_job_info]
+        assignees = [assignee for _, assignee in consensus_job_info]
+
+        consensus_job_data_providers = list(map(JobDataProvider, consensus_job_ids))
+        consensus_datasets = [
+            consensus_job_data_provider.dm_dataset
+            for consensus_job_data_provider in consensus_job_data_providers
+        ]
+
+        merged_dataset = merger(consensus_datasets)
+
+        assignee_report_data: Dict[User, Dict[str, Union[List[float], float]]] = {}
+
+        for idx, _ in enumerate(consensus_job_ids):
+            assignee_report_data.setdefault(assignees[idx], {})
+            assignee_report_data[assignees[idx]].setdefault("consensus_score", []).append(
+                merger._dataset_mean_consensus_score[id(consensus_datasets[idx])]
```

Comment:
```suggestion
                merger.dataset_mean_consensus_score[id(consensus_datasets[idx])]
```
Please refactor so that it wasn't a private field access.